go/vt/tabletserver/sqlquery.go

```diff
@@ -673,3 +672,30 @@ func withTimeout(ctx context.Context, timeout time.Duration) (context.Context, c
 	}
 	return context.WithTimeout(ctx, timeout)
 }
+
+func getColumnType(qre *QueryExecutor, columnName, tableName string) (int64, error) {
+	conn, err := qre.getConn(qre.qe.connPool)
+	if err != nil {
+		return mproto.VT_NULL, err
+	}
+	defer conn.Recycle()
+	query := fmt.Sprintf("SELECT %v FROM %v LIMIT 0", columnName, tableName)
```

Comment:
This would not help for this case unless we translate it into an AST . Change all manually constructed SQLs to AST representation would be a good idea.